### PR TITLE
Update libmesh, add clang container

### PR DESCRIPTION
**Pull request by milljm**
Install clang with dnf. Install matching gfortran version that coincides with the version of gcc that gets installed as a dependency from installing clang.

Set necessary /opt/rh/ toolchain paths. This was a key piece of information I missed as the dependencies above were being installed. `clang` gets installed to /usr/bin/, while the dependecies in order to properly make use of said clang are installed in /opt/rh/etc (some path that is not available by default like /usr/bin is). Hence, confusion.

Updates libmesh; summary of changes:

- Added low-order `RAVIART_THOMAS` finite element support, including
  general H(div) mapping support for future divergence-conforming
  finite element families.  Added `L2_RAVIART_THOMAS` support for the
  broken/L2 version of that space.  Example codes for each, using mixed
  and hybridized formulations.
- Option to remove a boundary id globally from a mesh without
  re-preparing it afterward
- Option to merge subdomains when stitching meshes based on inspection
  of both subdomain ids and subdomain names
- Better matching of input file ids when reading Abaqus files
- Fixes for independent control over which variables see p-refinement
- Added `QBase::clone()` method
- Added parameter for user-supplied hash when doing Just-In-Time
  compilation with fparser
- Added `Packing` specializations for `Eigen` matrices; this enables
  parallel communication of these classes and of many derived types
  including them.
- Bug fixes for nodal interpolation of rational elements; this
  corrects an error in VTK output of many IsoGeometric Analysis mesh
  results.
- Bug fix for overzealous caching of rational elements' shape
  functions.  This corrects an error in computation on many
  IsoGeometric Analysis meshes
- Bug fixes for refinement of Tet14 elements.
- More robust `reduced_basis` EIM error indicator calculations
- Higher-order default quadrature choice in integration-based error
  estimators and error indicators, avoiding underintegration on
  superconvergent results
- Removal of Fortran/C++ detection for HDF5 libraries, so we can more
  easily find the desired C HDF5 library we do use.
- Upgrade to latest autotools versions in libMesh, TIMPI, and
  MetaPhysicL
- Update VTK compatibility to support VTK 9.3
- GCC warning workarounds to allow --enable-werror builds with -Wextra
  in GCC 13
- Minor refactoring in Threads/PerfLog interactions
- More parallel synchronization assertions during System setup and
  update operations
- Minor fixes to documentation, comments, and unit tests, including
  VexCL-enabled unit tests of MetaPhysicL

### CODE REVIEW

**Comment by milljm** [comment]
Monday: Warnings being treated as errors:

https://civet.inl.gov/job/1861845/
https://civet.inl.gov/job/1861844/

**Comment by loganharbour** [comment]
Well @milljm, after cleaning up those warnings... I think this is actually ready to go. even passed the clang tests!

**Comment by milljm** [comment]
This is now an "everything" PR... lets pay double attention to all the meta.yaml, etc bumps going on before hitting merge...

**Comment by cticenhour** [comment]
How is 3a630b6 in this changeset not causing conflicts? Those changes are already in devel?

**Comment by roystgnr** [comment]
>How is https://github.com/idaholab/moose/commit/3a630b6df89896a83336cc382f57ebfa0d1b503e in this changeset not causing conflicts? Those changes are already in devel.

Git is smart about not double-applying the same commit.

File: scripts/tests/versioner_hashes.yaml [hunk 1]
```diff
@@ -153,3 +153,9 @@ d5e3e09267e1f2b47684a089150bec587178e1cf: #25784
   libmesh: 50463c4
   moose: 176f779
   wasp: 2fdee60
+286c130af84a0017140e6a5a910fb4bcfa47f8ae: #25831
+  mpic: e018192
```
Comment (milljm) [inline scripts/tests/versioner_hashes.yaml:157]: LOL...

**Comment by roystgnr** [comment]
C++ changes look good to me; I'm not confident enough in the apptainer/conda systems to hit approve on those changes.

If for some reason this doesn't get merged today I'll be pleading to advance the new libMesh hash to get a couple more upcoming changes there, but the libMesh update has been so delayed already that I don't want to delay it further on that account.

File: scripts/tests/versioner_hashes.yaml [hunk 2]
```diff
@@ -153,3 +153,9 @@ d5e3e09267e1f2b47684a089150bec587178e1cf: #25784
   libmesh: 50463c4
   moose: 176f779
   wasp: 2fdee60
+1cbf0084dc8ae0dde5f6a4500094f639eba55864: #25831
```
Comment (cticenhour) [inline scripts/tests/versioner_hashes.yaml:156]: Isn't this comment usually the PR number, not the referenced issue?

File: conda/pyhit/meta.yaml [hunk 3]
```diff
@@ -1,5 +1,4 @@
-{% set build = 0 %}
-{% set version = "2023.10.24" %}
+{% set version = "2023.11.06" %}
```
Comment (cticenhour) [inline conda/pyhit/meta.yaml:1]: `conda/pyhit/conda_build_config.yaml` still needs to be updated as well to `moose-wasp 2023.11.06`.

**Review by cticenhour — CHANGES_REQUESTED** [review]

File: conda/pyhit/meta.yaml (hunk 3 above)
Comment (milljm) [reply conda/pyhit/meta.yaml:1]: 👍  good catch!

File: framework/doc/packages_config.yml [hunk 4]
```diff
@@ -6,3 +6,4 @@ minimum_python: 3.7
 mpich: 4.0.2
 petsc_alt: 3.11.4
 petsc: 3.16.6
```
Comment (milljm) [inline framework/doc/packages_config.yml:8]: @cticenhour I should probably change this too!

Comment (cticenhour) [reply framework/doc/packages_config.yml:8]: Yes - we should while we're here. Good catch!

**Review by cticenhour — APPROVED** [review]
Review addressed. C++, conda, and documentation updates look reasonable to me. @loganharbour leaving one last look and any finishing requirements to you.

File: framework/doc/packages_config.yml (hunk 4 above)
Comment (cticenhour) [reply framework/doc/packages_config.yml]: Thanks for all the fixups @milljm!

**Comment by milljm** [comment]
LOL... with all the pushes, I done broke the PR Conda channel somehow. Fixing... then invalidate all...

**Comment by loganharbour** [comment]
We forgot about the fact that `moose-peacock` depends on `moose-mpich`. Bumped that. Version for moose didn't change, realized that we were also missing the influential files from `conda/peacock` for the `moose` version entry.

It never ends :-)

**Review by cticenhour — APPROVED** [review]

**Comment by loganharbour** [comment]
Failure is unrelated.

**Pull request merged**